updater/library/src/config.rs

```diff
@@ -0,0 +1,68 @@
+// This file handles the global config for the updater library.
+
+use std::cell::RefCell;
+
+use crate::updater::AppConfig;
+
+const DEFAULT_BASE_URL: &'static str = "https://shorebird-code-push-api-cypqazu4da-uc.a.run.app";
+const DEFAULT_CHANNEL: &'static str = "stable";
+
+thread_local!(static CONFIG: RefCell<Option<ResolvedConfig>> = RefCell::new(None));
+
+pub fn with_config<F, R>(f: F) -> R
+where
+    F: FnOnce(&ResolvedConfig) -> R,
```

Comment:
This is a type constraint.  Its saying that "F" is a kind of function/lambda (FnOnce, they're ephemeral) that takes this specific signature of a reference/borrow to a ResolveConfig and returns R (another type).  Where clauses like this are common for dealing with long types in the signature which may need to be repeated (as this one is).